hal/network/lwip/cellular/pppncpnetif.cpp

```diff
@@ -150,41 +165,61 @@ void PppNcpNetif::loop(void* arg) {
 }
 
 int PppNcpNetif::up() {
-    NetifEvent ev = NetifEvent::Up;
-    int r = os_queue_put(queue_, &ev, 0, nullptr);
-    if (r) {
-        // FIXME: we'll miss this event, just as an attempt to potentially recover
-        // set internal state.
-        up_ = true;
+    if (lastNetifEvent_ == NetifEvent::Up) {
+        return SYSTEM_ERROR_NONE;
     }
-    return r;
+    lastNetifEvent_ = NetifEvent::Up;
+    setExpectedInternalState(lastNetifEvent_);
```

Comment:
Note that three separate sequential atomic operations do not make the whole change atomic. You may end up in an inconsistent state. This needs to be a single `struct` for example and there should be a single modification or we should rely on the NCP thread to at least perform the other two modifications based on the posted event.

This applies to all the other cases.

It's probably fine in our particular use case as all of these events will be coming from system thread though.